lib/vector-buffers/src/buffer_usage_data.rs

```diff
@@ -215,10 +234,15 @@ impl BufferUsage {
                         byte_size: stage.sent_byte_size.swap(0, Ordering::Relaxed),
                     });
 
-                    if let Some(dropped_event_count) = &stage.dropped_event_count {
+                    if let Some(dropped_event_data) = &stage.dropped_event_data {
                         emit(EventsDropped {
                             idx: stage.idx,
-                            count: dropped_event_count.swap(0, Ordering::Relaxed),
+                            count: dropped_event_data
+                                .dropped_event_count
+                                .swap(0, Ordering::Relaxed),
+                            byte_size: dropped_event_data
+                                .dropped_event_size
+                                .swap(0, Ordering::Relaxed),
```

Comment:
This looks like this would be racy. Does is matter that the `count` and `byte_size` could be updated between these two fetches?

Reply:
No more than the other metrics here, I don't think.

Reply:
Fair enough.